extensions/azure/data-plane/data-factory/src/test/java/org/eclipse/dataspaceconnector/azure/dataplane/azuredatafactory/AzureDataFactoryCopyIntegrationTest.java

```diff
@@ -85,6 +93,7 @@ static void beforeAll() throws FileNotFoundException {
     static void afterAll() {
         System.setProperties(savedProperties);
         containerCleanup.parallelStream().forEach(Runnable::run);
+        secretCleanup.forEach(Runnable::run);
```

Comment:
why not parallel stream?
In addition, consider merging `containerCleanup` and `secretCleanup` for simplicity and perf

Reply:
Because for secrets it needs to delete it first and then purge it. Hence non-parallel.

Reply:
I mean why are secrets not deleted in parallel?